### PR TITLE
Search movies

### PR DESCRIPTION
**Pull request by marianasmmattos**
https://reactjs-search-movies.netlify.app/

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import ResultsList from './containers/resultsList';
+
+function App() {
+  const [ state, setState ] = useState({
+    searchInput: "",
+    content: <Welcome />,
+    selectedResult: {}
+  });
+
+  const baseURL = "https://api.themoviedb.org/3/search/movie?api_key=54d5d6e26602b36d146edeb90b272ae5&language=pt-br&";
```
Comment (danonel) [inline src/App.js:17]: You can create an axios instance with axios.create to pass the baseUrl and others configs

File: src/components/emptyList/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import './index.css';
+
+function EmptyList() {
+  const sadFaceImage = '/images/sad_face.png';
+
+  const emptyMessageTitle = 'Sorry! No results found';
+  const emptyMessageSubtitle = 'What about searching for another movie?'
```
Comment (danonel) [inline src/components/emptyList/index.js:8]: You can recive the empty message title/subtitle as props. And you can add a default value to this props

File: src/containers/welcome/index.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import './index.css';
+
+function Welcome() {
+  const welcomeTitle = '🎉 Welcome! 🎉';
+  const welcomeSubtitle = 'Here you can find any movie you want, start typing above'
```
Comment (danonel) [inline src/containers/welcome/index.js:6]: You can recive the welcome message title/subtitle as props. And you can add a default value to this props

**Review by danonel — APPROVED** [review]
LGTM

File: src/App.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import Search from './components/search/search';
+import Welcome from './containers/welcome';
+import ResultsList from './containers/resultsList';
+
+function App() {
+  const [ state, setState ] = useState({
+    searchInput: "",
+    content: <Welcome />,
+    selectedResult: {}
+  });
```
Comment (marianasmmattos) [inline src/App.js:15]:
```suggestion
  const [ state, setState ] = useState({
    searchInput: "",
    welcome: true,
    movieList: [],
    selectedResult: {}
  });
```

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    setState(prevState => {
+      return { ...prevState, searchInput: e.target.value }
+    });
+  }
+
+  const search = (e) => {
+    if(e.key === "Enter"){
+      axios(`${baseURL}query=${state.searchInput}`).then(({ data }) => {
+        setState(prevState => {
+          return { ...prevState, content: <ResultsList movieList={data.results} /> }
```
Comment (marianasmmattos) [inline src/App.js:29]:
```suggestion
        setState(prevState => ({ ...prevState, welcome: false, movieList: data.results /> })
```

File: src/App.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  });
+
+  const baseURL = "https://api.themoviedb.org/3/search/movie?api_key=54d5d6e26602b36d146edeb90b272ae5&language=pt-br&";
+  
+  const handleInput = (e) => {
+    setState(prevState => {
+      return { ...prevState, searchInput: e.target.value }
+    });
+  }
+
```
Comment (marianasmmattos) [inline src/App.js:24]:
```suggestion
const hasSearch = !state.welcome;
```

File: src/App.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  return (
+    <div className="App">
+      <header>
+        <h1>Search your movies</h1>
+      </header>
+      <main>
+        <Search handleInput={handleInput} search={search} />
+        {state.content}
```
Comment (marianasmmattos) [inline src/App.js:42]:
```suggestion
        <Search handleInput={handleInput} search={search} />
        { hasSearch ? <ResultsList movieList={state.movieList} /> : <Welcome />}
```

**Pull request closed without merging**